### PR TITLE
Include group info on Observation view

### LINKED ISSUE #439: Include parent information on groupEdit subscription
This is so we can update Explore correctly when, for example, a group is moved. Both `parentGroup` and `parentGroupIndex` would be need to be included

### LINKED ISSUE #440: Include group info on observationEdit subscription
To move observations when they are moved between/outside groups in Explore. Both `groupId` and `groupIndex` would need to be included

### PR DESCRIPTION
**Pull request by hugo-vrijswijk**
Fixes #440

Adds group info to the `Observation` type

### CODE REVIEW

File: modules/schema/src/main/resources/lucuma/odb/graphql/OdbSchema.graphql [hunk 1]
```diff
@@ -7137,6 +7137,16 @@ type Observation {
...
 
+  """
+  Enclosing group, if any.
+  """
+  groupId: GroupId
+
+  """
+  Index in enclosing group or at the top level if ungrouped. If left unspecified on creation, observation will be added last in its enclosing group or at the top level. Cannot be set to null.
+  """
+  groupIndex: NonNegShort
```
Comment (rpiaggio) [inline modules/schema/src/main/resources/lucuma/odb/graphql/OdbSchema.graphql:7148]: Should this be `NonNegShort!` if it can't be null?

Comment (hugo-vrijswijk) [reply modules/schema/src/main/resources/lucuma/odb/graphql/OdbSchema.graphql:7148]: Hmm, maybe? They are nullable in the database, but none of the rows on staging are.  I copied this over from another part of the schema.

image

image

It looks like `groupIndex` is nonnullable? @tpolecat ?

Comment (tpolecat) [reply modules/schema/src/main/resources/lucuma/odb/graphql/OdbSchema.graphql:7148]: Should be `NOT NULL`. It starts off `NULL` here https://github.com/gemini-hlsw/lucuma-odb/blob/main/modules/service/src/main/resources/db/migration/V0290__groups.sql#L39 but it's set to `NOT NULL` at line 57.

Comment (tpolecat) [reply modules/schema/src/main/resources/lucuma/odb/graphql/OdbSchema.graphql:7148]:
```
lucuma-odb=# \d t_observation
                                                                                                                                                                                                                        Table "public.t_observation"
           Column           |            Type             | Collation | Nullable |                                                                                                                                                                                          Default                                                                                                                                                                                          
----------------------------+-----------------------------+-----------+----------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 c_program_id               | d_program_id                |           | not null | 
...
 c_group_id                 | d_group_id                  |           |          | 
 c_group_index              | smallint                    |           | not null | 
...
```

Comment (hugo-vrijswijk) [reply modules/schema/src/main/resources/lucuma/odb/graphql/OdbSchema.graphql]: Thanks, fixed (and in the other place it was used too)

**Comment by hugo-vrijswijk** [comment]
> Looks good so far. Can you add these fields to some of the tests (for obs creation maybe? group manipulation?) to ensure that they're selected properly?

Added, I wasn't sure where the tests were at first 😅

**Review by tpolecat — APPROVED** [review]

**Pull request merged**